www-aps/qtchan/patches/qtchan-0.95.patchset

```diff
@@ -0,0 +1,60 @@
+From 19fb61ef810b8356cd77f5a5720b439020a0fc0e Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Zolt=C3=A1n=20Mizsei?= <zmizsei@extrowerk.com>
+Date: Tue, 1 May 2018 16:36:05 +0200
+Subject: Do not create dot folders in $HOME
+
+
+diff --git a/mainwindow.cpp b/mainwindow.cpp
+index 8990d2e..16beea8 100644
+--- a/mainwindow.cpp
++++ b/mainwindow.cpp
+@@ -63,7 +63,11 @@ MainWindow::MainWindow(QWidget *parent) :
+ 	connect(&settingsView,&Settings::update,[=](QString field, QVariant value){
+ 		if(field == "use4chanPass" && value.toBool() == true){
+ 			QSettings settings(QSettings::IniFormat,QSettings::UserScope,"qtchan","qtchan");
+-			QString defaultCookies = QDir::homePath() + "/.config/qtchan/cookies";
++			#ifndef __HAIKU__
++				QString defaultCookies = QDir::homePath() + "/.config/qtchan/cookies";
++			#else
++				QString defaultCookies = QDir::homePath() + "/config/settings/qtchan/cookies";
```

Comment:
Shouldn't [`find_directory()`][0] be used instead of hardcoding paths? Also, for cookies, I think `B_USER_CACHE_DIRECTORY` or `B_USER_VAR_DIRECTORY` would be a better place to put them into

[0]: https://www.haiku-os.org/docs/api/FindDirectory_8h.html#af95c55d472448ab7af9f32c33aeea71e

Reply:
Yep, it would be nice to use `find_directory()`.

Regarding the cookies, I'd say that it is OK (and even better imo) to keep them in `~/config/settings/qtchan/cookies/`.
Keeping them in `B_USER_CACHE_DIRECTORY` or `B_USER_VAR_DIRECTORY` would mean that other programs would possibly notice them, while having them in a dedicated directory (like the one we have right now in the recipe) is a little bit safer.

Reply:
Oh, I meant `B_USER_CACHE_DIRECTORY/qtchan/cookies` instead.

Reply:
Well, even if it's in a subdirectory of `B_USER_CACHE_DIRECTORY` I would still think that the cookies would be better in `~/config/settings/qtchan/cookies/`.
My point of view is that cookies are not really cache data. They are user data created by an application.
Cached data, on the other hand, can be restored from an external source. Cookies can only be "recreated" (but not always restored to what the user had previously), provided the user submits the appropriate inputs, e.g. credentials.

Now, if more contributors think `B_USER_CACHE_DIRECTORY` would be better than `~/config/settings/qtchan/cookies/` then I'm OK with your suggestion :)

Reply:
I've also looked at WebPositive [source][0] and found that cookies are stored in `B_USER_SETTINGS_DIRECTORY`, so I agree that's where we should store the cookies.

[0]Y http://xref.plausible.coop/source/xref/haiku/src/apps/webpositive/BrowserApp.cpp#90